src/components/fields/DataSelector.js

```diff
@@ -44,7 +44,17 @@ class DataSelector extends Component {
     }
     const update = {};
     if (this.dataSrcExists) {
-      update[this.srcAttr] = value;
+      const srcAttr = this.srcAttr;
+      update[srcAttr] = value;
+
+      if (!value) {
+        /*
```

Comment:
I don't know if you saw my comments on this block, but I think it can just be folded into the next one... basically just remove this if and the next and replace them with just `update[this.props.attr] = this.dataSources[value]`.

This way, if `value` is falsey or otherwise not contained in the datasource, the `attr` is automatically cleared, and we don't need to do string manipulation on `srcAttr`.